src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -1,9 +1,93 @@
 package uk.gov.ons.ctp.integration.contactcentresvc.service.impl;
 
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.UUID;
+import ma.glasnost.orika.CustomConverter;
+import ma.glasnost.orika.MapperFacade;
+import ma.glasnost.orika.MapperFactory;
+import ma.glasnost.orika.MappingContext;
+import ma.glasnost.orika.converter.ConverterFactory;
+import ma.glasnost.orika.impl.DefaultMapperFactory;
+import ma.glasnost.orika.metadata.Type;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
 import org.springframework.validation.annotation.Validated;
+import org.springframework.web.server.ResponseStatusException;
+import uk.gov.ons.ctp.common.time.DateTimeUtil;
+import uk.gov.ons.ctp.integration.contactcentresvc.client.caseservice.CaseServiceClientServiceImpl;
+import uk.gov.ons.ctp.integration.contactcentresvc.client.caseservice.model.CaseContainerDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.CaseDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.CaseRequestDTO;
 import uk.gov.ons.ctp.integration.contactcentresvc.service.CaseService;
+import uk.gov.ons.ctp.integration.contactcentresvc.utility.Constants;
 
 @Service
 @Validated()
-public class CaseServiceImpl implements CaseService {}
+public class CaseServiceImpl implements CaseService {
+  private static final Logger log = LoggerFactory.getLogger(CaseServiceImpl.class);
+
+  private MapperFacade caseDTOMapper;
+
+  @Autowired private CaseServiceClientServiceImpl caseServiceClient;
+
+  private static class DateMapper extends CustomConverter<String, Date> {
+    private SimpleDateFormat caseDateTimeFormatter =
```

Comment:
SimpleDateFormat is not thread safe. This is one instance...

Reply:
I think that you've reviewed an older version of the code, as this code was deleted when I updated the CaseService DTO's to use 'Date' instead of just 'String'. (I must confess to missing that at the time though)

I've searched the rest of the code and the only other uses of SimpleDateFormatter are all thread local (except possibly for the pre-existing MultiIsoDateFormat class).